### DISCUSSION TITLE
Distributed file copying in a LAN

### DISCUSSION THREAD

**Discussion by brianjmurrell** [rakshasa/rtorrent#1320] [Q&A]
I have a situation where many computers on my LAN can want a copy of the same large file from a server at the same time.  Right now they just all initiate their own 1:1 copy and you can guess what happens next: many parallel copies start at "around" the same time and saturate the network and disk of the server.

This seems like a problem that torrenting is ripe to solve -- where the clients all start requesting different portions of the file from the server and then start sharing the portions they have with each other.

Is this something that rtorrent can handle?  I want to be sure the file is only being shared among the clients on my network and don't end up sharing with torrent clients outside of my network of course.

I know very little of the details of how torrenting works and trackers and what-not so any pointers to how rtorrent could solve this problem would be greatly appreciated.

Interestingly, somebody else asking a similar/the same [question on reddit](https://www.reddit.com/r/rtorrent/comments/1gjeetv/create_and_seed_torrent_for_lan_only_python/).

**Comment by rakshasa** [comment]
This isn't the right place for user support.
